From c8197e1e231df7f71007de69a3bd362ec4f96317 Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Fri, 5 Feb 2016 17:51:05 +0000
Subject: [PATCH] [networking] blacklist Cisco CDP paths

Cisco CDP products include a kernel module that presents several
large pseudo files in /proc and /sys. Blacklist these locations
to avoid attempting to read the content into reports.

Fixes #740.

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/networking.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sos/plugins/networking.py b/sos/plugins/networking.py
index 45a4c02..dd377b4 100644
--- a/sos/plugins/networking.py
+++ b/sos/plugins/networking.py
@@ -128,6 +128,9 @@ class Networking(Plugin):
         self.add_forbidden_path("/proc/net/rpc/use-gss-proxy")
         self.add_forbidden_path("/proc/net/rpc/*/channel")
         self.add_forbidden_path("/proc/net/rpc/*/flush")
+        # Cisco CDP
+        self.add_forbidden_path("/proc/net/cdp")
+        self.add_forbidden_path("/sys/net/cdp")
 
         self.add_cmd_output("ip -o addr", root_symlink="ip_addr")
         self.add_cmd_output("route -n", root_symlink="route")
-- 
1.8.3.1

